base/uk.ac.stfc.isis.ibex.e4.ui.perspectiveswitcher/src/uk/ac/stfc/isis/ibex/e4/ui/perspectiveswitcher/controls/ButtonViewModel.java

```diff
@@ -98,4 +105,39 @@ public String getText() {
     protected void setText(String newText) {
         firePropertyChange("text", text, text = newText);
```

Comment:
If you do the logic of adding the spacing on and how to behave different for maximised/minimised in here you won't have to do it everywhere else where the text is changed

Reply:
I think there's still some additional logic you can put in here about what to do on minimised. If minimised store the text you would have set and instead set "". This simplifies the minimise/maximise methods and means that callers of `setText` need not have to worry about whether the button is maximised or not.